src/reusable-blocks/class-reusable-blocks-basket-handler.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace WPML\PB\Gutenberg;
+
+class Reusable_Blocks_Basket_Handler extends Reusable_Blocks_Handler {
+
+	/** @var \WPML_Translation_Basket $translation_basket */
+	private $translation_basket;
+
+	public function __construct(
+		Reusable_Blocks $reusable_blocks,
+		Reusable_Blocks_Translation $reusable_blocks_translation,
+		\WPML_Translation_Basket $translation_basket
+	) {
+		parent::__construct( $reusable_blocks, $reusable_blocks_translation );
+		$this->translation_basket = $translation_basket;
+	}
+
+	/**
+	 * @param array $data
+	 */
+	public function add_blocks( array $data ) {
+		if ( ! isset( $data['post'], $data['translate_from'], $data['tr_action'] ) ) {
+			return;
+		}
+
+		$post_elements = $this->extract_added_post_elements( $data );
+		$blocks        = $this->get_blocks_from_post_elements( $post_elements );
+
+		$blocks = $blocks->map( [ $this, 'select_target_langs' ] )
+			->filter( function( $block ) { return ! empty( $block['target_langs'] ); } )
+			->toArray();
+
+		if ( $blocks ) {
+			$basket_portion = [
+				'post'              => [],
+				'source_language'	=> $data['translate_from'],
+				'target_languages'	=> array_keys( $data['tr_action'] ),
+			];
+
+			foreach ( $blocks as $block ) {
```

Comment:
I started writing this piece of code in the functional way, but there's a bug with `mapWithKeys` in version 5.3 (https://github.com/laravel/framework/issues/15409), it won't work with numeric keys.